test/e2e/bootstrapper_test.go

```diff
@@ -125,26 +133,33 @@ func svcExists(t *testing.T, svcName string) bool {
 	}
 }
 
-// getSvcState gets the current state of the specified service. Requires administrator privileges
-func getSvcState(svcName string) (svc.State, error) {
+// getSvcInfo gets the current state and the fully qualified path of the specified service.
+// Requires administrator privileges
+func getSvcInfo(svcName string) (svc.State, string, error) {
 	// State(0) is equivalent to "Stopped"
 	state := svc.State(0)
 	svcMgr, err := mgr.Connect()
 	if err != nil {
-		return state, fmt.Errorf("could not connect to Windows SCM: %s", err)
+		return state, "", fmt.Errorf("could not connect to Windows SCM: %s", err)
 	}
 	defer svcMgr.Disconnect()
 	mySvc, err := svcMgr.OpenService(svcName)
 	if err != nil {
 		// Could not find the service, so it was never created
-		return state, err
+		return state, "", err
 	}
 	defer mySvc.Close()
+	// Get state of Service
 	status, err := mySvc.Query()
 	if err != nil {
-		return state, err
+		return state, "", err
+	}
+	// Get fully qualified path of Service
+	config, err := mySvc.Config()
```

Comment:
Can this be empty or nil?

Reply:
Even if the err is not nil, the most this function will do is return empty string for BinaryPathName. I think I will add an additional check in the caller function to make sure Path is not empty before proceeding.

Reply:
The empty path check should happen here and not in the call site. You need to give some guarantee to the caller that the path will be valid if there is no error present.